File: 6-kyu/validate-credit-card-number/python/solution.py
def validate(n):
    n_arr = [int(digit) for digit in str(n)]
    for i in range(len(n_arr) - 2, -1, -2):
        n_arr[i] *= 2
        if n_arr[i] > 9:
            n_arr[i] -= 9
    return sum(n_arr) % 10 == 0

File: 6-kyu/validate-credit-card-number/c++/solution.cpp
#include <algorithm>
#include <numeric>
#include <vector>

class Kata {
  public:
  static bool validate(long long int n) {
    auto tokens = tokenize(n);
    for (unsigned i = 1; i < tokens.size(); i += 2) {
      tokens[i] *= 2;
      if (tokens[i] > 9) tokens[i] -= 9;
    }
    auto sum = std::accumulate(tokens.begin(), tokens.end(), 0);
    return sum % 10 == 0;
  }
  private:
  static std::vector<long long> tokenize(long long int n) {
    std::vector<long long> ret;
    while (n > 0) {
      ret.push_back(n % 10);
      n /= 10;
    }
    return ret;
  }
};